src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -776,23 +776,19 @@ public byte[] ToArray()
 
             private void EnsureCapacity(int value)
             {
-                if (value > _buffer.Length)
+                if ((uint)value > (uint)_maxBufferSize) // value cast handles overflow to negative as well
                 {
-                    Grow(value);
+                    throw CreateOverCapacityException(_maxBufferSize);
                 }
-                else if (value < 0) // overflow
+                else if (value > _buffer.Length)
                 {
-                    throw CreateOverCapacityException(_maxBufferSize);
+                    Grow(value);
                 }
             }
 
             private void Grow(int value)
             {
                 Debug.Assert(value > _buffer.Length);
-                if (value > _maxBufferSize)
```

Comment:
We could.  I'd left it out because it's not actually important to the algorithm for growing; the caller wants to increase the array size to one large enough to hold the new size, and the function does that.  It's the callers job to decide if such growth is appropriate or not.  At least that was my reasoning.